Index: configure
===================================================================
--- configure	(revision 29343)
+++ configure	(working copy)
@@ -60,6 +60,7 @@
 drascula
 gob
 igor
+kom
 kyra
 lure
 parallaction
@@ -93,6 +94,8 @@
 _engine_gob_build=yes
 _engine_igor_name="Igor: Objective Uikokahonia"
 _engine_igor_build=no
+_engine_kom_name="Kingdom O' Magic"
+_engine_kom_build=no
 _engine_kyra_name="Legend of Kyrandia"
 _engine_kyra_build=yes
 _engine_lure_name="Lure of the Temptress"
Index: engines/engines.mk
===================================================================
--- engines/engines.mk	(revision 29343)
+++ engines/engines.mk	(working copy)
@@ -55,6 +55,12 @@
 MODULES += engines/igor
 endif
 
+ifdef DISABLE_KOM
+DEFINES += -DDISABLE_KOM
+else 
+MODULES += engines/kom
+endif
+
 ifdef DISABLE_KYRA
 DEFINES += -DDISABLE_KYRA
 else
Index: base/plugins.cpp
===================================================================
--- base/plugins.cpp	(revision 29343)
+++ base/plugins.cpp	(working copy)
@@ -110,6 +110,9 @@
 		#ifndef DISABLE_IGOR
 		LINK_PLUGIN(IGOR)
 		#endif
+		#ifndef DISABLE_KOM
+		LINK_PLUGIN(KOM)
+		#endif
 		#ifndef DISABLE_KYRA
 		LINK_PLUGIN(KYRA)
 		#endif
