Index: configure
===================================================================
--- configure	(revision 27697)
+++ configure	(working copy)
@@ -63,6 +63,7 @@
 _build_touche=yes
 _build_parallaction=yes
 _build_cruise=no
+_build_kom=no
 _build_hq_scalers=yes
 _build_scalers=yes
 
@@ -367,6 +368,7 @@
   --disable-sword2         don't build the Broken Sword 2 engine
   --disable-touche         don't build the Touche: The Adventures of the Fifth Musketeer engine
 
+  --enable-kom             build the Kingdom O' Magic engine
   --enable-plugins         build engines as loadable modules instead of
                            static linking them
   --disable-mt32emu        don't enable the integrated MT-32 emulator
@@ -438,6 +440,7 @@
       --disable-touche)         _build_touche=no ;;
       --disable-parallaction)   _build_parallaction=no ;;
       --enable-cruise)          _build_cruise=yes ;;
+      --enable-kom)             _build_kom=yes ;;
       --disable-hq-scalers)     _build_hq_scalers=no ;;
       --disable-scalers)        _build_scalers=no ;;
       --enable-alsa)            _alsa=yes       ;;
@@ -730,6 +733,7 @@
 add_to_config_mk_if_no $_build_touche       'DISABLE_TOUCHE = 1'
 add_to_config_mk_if_no $_build_parallaction 'DISABLE_PARALLACTION = 1'
 add_to_config_mk_if_no $_build_cruise       'DISABLE_CRUISE = 1'
+add_to_config_mk_if_no $_build_kom          'DISABLE_KOM = 1'
 add_to_config_mk_if_no $_build_hq_scalers   'DISABLE_HQ_SCALERS = 1'
 add_to_config_mk_if_no $_build_scalers      'DISABLE_SCALERS = 1'
 
@@ -1386,6 +1390,9 @@
 if test "$_build_cruise" = yes ; then
 	echo "    Cinematique evo 2"
 fi
+if test "$_build_kom" = yes ; then
+	echo "    Kingdom O' Magic"
+fi
 
 echo
 
Index: engines/engines.mk
===================================================================
--- engines/engines.mk	(revision 27697)
+++ engines/engines.mk	(working copy)
@@ -96,3 +96,9 @@
 else
 MODULES += engines/cruise
 endif
+
+ifdef DISABLE_KOM
+DEFINES += -DDISABLE_KOM
+else 
+MODULES += engines/kom
+endif
Index: base/plugins.cpp
===================================================================
--- base/plugins.cpp	(revision 27697)
+++ base/plugins.cpp	(working copy)
@@ -131,6 +131,9 @@
 		#ifndef DISABLE_CRUISE
 		LINK_PLUGIN(CRUISE)
 		#endif
+		#ifndef DISABLE_KOM
+		LINK_PLUGIN(KOM)
+		#endif
 
 		return pl;
 	}
