Index: base/plugins.cpp
===================================================================
--- base/plugins.cpp	(revision 31898)
+++ base/plugins.cpp	(working copy)
@@ -123,6 +123,9 @@
 		#if PLUGIN_ENABLED_STATIC(IGOR)
 		LINK_PLUGIN(IGOR)
 		#endif
+		#if PLUGIN_ENABLED_STATIC(KOM)
+		LINK_PLUGIN(KOM)
+		#endif
 		#if PLUGIN_ENABLED_STATIC(KYRA)
 		LINK_PLUGIN(KYRA)
 		#endif
Index: configure
===================================================================
--- configure	(revision 31898)
+++ configure	(working copy)
@@ -90,6 +90,7 @@
 add_engine drascula "Drascula: The Vampire Strikes Back" yes
 add_engine gob "Gobli*ns" yes
 add_engine igor "Igor: Objective Uikokahonia" no
+add_engine kom "Kingdom O' Magic" no
 add_engine kyra "Legend of Kyrandia" yes
 add_engine lure "Lure of the Temptress" yes
 add_engine m4 "M4/MADS" no
Index: engines/engines.mk
===================================================================
--- engines/engines.mk	(revision 31898)
+++ engines/engines.mk	(working copy)
@@ -47,6 +47,11 @@
 MODULES += engines/igor
 endif
 
+ifdef ENABLE_KOM
+DEFINES += -DENABLE_KOM=$(ENABLE_KOM)
+MODULES += engines/kom
+endif
+
 ifdef ENABLE_KYRA
 DEFINES += -DENABLE_KYRA=$(ENABLE_KYRA)
 MODULES += engines/kyra
