Index: configure
===================================================================
--- configure	(revision 28139)
+++ configure	(working copy)
@@ -64,6 +64,7 @@
 _build_parallaction=yes
 _build_cruise=no
 _build_drascula=no
+_build_kom=no
 _build_hq_scalers=yes
 _build_scalers=yes
 
@@ -369,6 +370,7 @@
   --disable-sword2         don't build the Broken Sword 2 engine
   --disable-touche         don't build the Touche: The Adventures of the Fifth Musketeer engine
 
+  --enable-kom             build the Kingdom O' Magic engine
   --enable-plugins         build engines as loadable modules instead of
                            static linking them
   --disable-mt32emu        don't enable the integrated MT-32 emulator
@@ -441,6 +443,7 @@
       --disable-parallaction)   _build_parallaction=no ;;
       --enable-cruise)          _build_cruise=yes ;;
       --enable-drascula)        _build_drascula=yes ;;
+      --enable-kom)             _build_kom=yes ;;
       --disable-hq-scalers)     _build_hq_scalers=no ;;
       --disable-scalers)        _build_scalers=no ;;
       --enable-alsa)            _alsa=yes       ;;
@@ -734,6 +737,7 @@
 add_to_config_mk_if_no $_build_parallaction 'DISABLE_PARALLACTION = 1'
 add_to_config_mk_if_no $_build_cruise       'DISABLE_CRUISE = 1'
 add_to_config_mk_if_no $_build_drascula     'DISABLE_DRASCULA = 1'
+add_to_config_mk_if_no $_build_kom          'DISABLE_KOM = 1'
 add_to_config_mk_if_no $_build_hq_scalers   'DISABLE_HQ_SCALERS = 1'
 add_to_config_mk_if_no $_build_scalers      'DISABLE_SCALERS = 1'
 
@@ -1395,6 +1399,9 @@
 if test "$_build_drascula" = yes ; then
 	echo "    Drascula"
 fi
+if test "$_build_kom" = yes ; then
+	echo "    Kingdom O' Magic"
+fi
 
 echo
 
Index: engines/engines.mk
===================================================================
--- engines/engines.mk	(revision 28139)
+++ engines/engines.mk	(working copy)
@@ -102,3 +102,9 @@
 else
 MODULES += engines/drascula
 endif
+
+ifdef DISABLE_KOM
+DEFINES += -DDISABLE_KOM
+else 
+MODULES += engines/kom
+endif
Index: base/plugins.cpp
===================================================================
--- base/plugins.cpp	(revision 28139)
+++ base/plugins.cpp	(working copy)
@@ -134,6 +134,9 @@
 		#ifndef DISABLE_DRASCULA
 		LINK_PLUGIN(DRASCULA)
 		#endif
+		#ifndef DISABLE_KOM
+		LINK_PLUGIN(KOM)
+		#endif
 
 		return pl;
 	}
