Index: base/plugins.cpp
===================================================================
--- base/plugins.cpp	(revision 52170)
+++ base/plugins.cpp	(working copy)
@@ -118,6 +118,9 @@
 		#if PLUGIN_ENABLED_STATIC(HUGO)
 		LINK_PLUGIN(HUGO)
 		#endif
+		#if PLUGIN_ENABLED_STATIC(KOM)
+		LINK_PLUGIN(KOM)
+		#endif
 		#if PLUGIN_ENABLED_STATIC(KYRA)
 		LINK_PLUGIN(KYRA)
 		#endif
Index: configure
===================================================================
--- configure	(revision 52170)
+++ configure	(working copy)
@@ -89,6 +89,7 @@
 add_engine groovie "Groovie" yes "groovie2"
 add_engine groovie2 "Groovie 2 games" no
 add_engine hugo "Hugo Trilogy" no
+add_engine kom "Kingdom O' Magic" no
 add_engine kyra "Legend of Kyrandia" yes "lol"
 add_engine lol "Lands of Lore" no
 add_engine lure "Lure of the Temptress" yes
Index: engines/engines.mk
===================================================================
--- engines/engines.mk	(revision 52170)
+++ engines/engines.mk	(working copy)
@@ -65,6 +65,11 @@
 MODULES += engines/hugo
 endif
 
+ifdef ENABLE_KOM
+DEFINES += -DENABLE_KOM=$(ENABLE_KOM)
+MODULES += engines/kom
+endif
+
 ifdef ENABLE_KYRA
 DEFINES += -DENABLE_KYRA=$(ENABLE_KYRA)
 MODULES += engines/kyra
