Index: configure
===================================================================
--- configure	(revision 29413)
+++ configure	(working copy)
@@ -69,6 +69,7 @@
 add_engine cruise "Cinematique evo 2" no
 add_engine drascula "Drascula: The Vampire Strikes Back" no
 add_engine gob "Gobli*ns" yes
+add_engine kom "Kingdom O' Magic" no
 add_engine igor "Igor: Objective Uikokahonia" no
 add_engine kyra "Legend of Kyrandia" yes
 add_engine lure "Lure of the Temptress" yes
Index: engines/engines.mk
===================================================================
--- engines/engines.mk	(revision 29413)
+++ engines/engines.mk	(working copy)
@@ -55,6 +55,12 @@
 MODULES += engines/igor
 endif
 
+ifdef DISABLE_KOM
+DEFINES += -DDISABLE_KOM
+else 
+MODULES += engines/kom
+endif
+
 ifdef DISABLE_KYRA
 DEFINES += -DDISABLE_KYRA
 else
Index: base/plugins.cpp
===================================================================
--- base/plugins.cpp	(revision 29413)
+++ base/plugins.cpp	(working copy)
@@ -110,6 +110,9 @@
 		#ifndef DISABLE_IGOR
 		LINK_PLUGIN(IGOR)
 		#endif
+		#ifndef DISABLE_KOM
+		LINK_PLUGIN(KOM)
+		#endif
 		#ifndef DISABLE_KYRA
 		LINK_PLUGIN(KYRA)
 		#endif
